packages/strapi-utils/lib/sanitize-entity.js

```diff
@@ -120,13 +121,23 @@ const getNextFields = (fields, key, { allowedFieldsHasKey }) => {
   return [nextFields, isAllowed];
 };
 
-const shouldRemoveAttribute = (attribute, { withPrivate, isOutput }) => {
-  if (_.isNil(attribute)) {
-    return false;
-  }
+const getPrivateAttributes = model => {
+  return _.difference(
+    _.union(
+      strapi.config.get('api.responses.privateAttributes', []),
+      _.get(model, 'options.privateAttributes', [])
+    ),
+    _.get(model, 'options.ignorePrivateFor', [])
+  );
+};
 
+const shouldRemoveAttribute = (
+  key,
+  attribute = {},
+  { withPrivate, isOutput, privateAttributes }
+) => {
   const isPassword = attribute.type === 'password';
-  const isPrivate = attribute.private === true;
+  const isPrivate = attribute.private === true || privateAttributes.includes(key);
```

Comment:
What about when `attribute.private === true` but the attribute has been ignored in `ignorePrivateFor` array? Here it would set the `isPrivate` boolean to true (because of the `||`), but I would expect it to be set to false (because it has been explicitly ignored)

Reply:
Oh, I thought `attribute.private` has more priority over `privateAttributes`.